lib/secp256k1/schnorr.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Bitcoinex.Secp256k1.Schnorr do
+  @moduledoc """
+  Schnorr-specific secp256k1 operations
+  """
+  alias Bitcoinex.Secp256k1
+  alias Bitcoinex.Secp256k1.{Math, Params, Point, PrivateKey, Signature}
+  alias Bitcoinex.Utils
+
+  @n Params.curve().n
+
+  @generator_point %Point{
+    x: Params.curve().g_x,
+    y: Params.curve().g_y
+  }
+
+  @spec sign(PrivateKey.t(), non_neg_integer(), non_neg_integer()) ::
```

Comment:
should we move aux to an optional argument(maybe a option list?) as I believe most of the time user don't want to bother about generating a random aux in their code?